tools/ci-scripts/linux/image-builder/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:16.04
+FROM ubuntu:18.04
```

Comment:
docker succeeds in building image, but fails to build bond with gcc.

Reply:
`[ 15%] Building CXX object cpp/test/compat/core/CMakeFiles/compatibility_test.dir/compat.cpp.o
[ 15%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/validation_types.cpp.o
[ 15%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/scope_test1_types.cpp.o
[ 16%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/scope_test2_types.cpp.o
[ 16%] Building CXX object cpp/test/compat/core/CMakeFiles/compatibility_test.dir/serialization.cpp.o
[ 16%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/cmdargs_types.cpp.o
[ 16%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/unit_test_core_apply.cpp.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
make[3]: *** [cpp/test/compat/core/CMakeFiles/compatibility_test.dir/serialization.cpp.o] Error 4
cpp/test/compat/core/CMakeFiles/compatibility_test.dir/build.make:167: recipe for target 'cpp/test/compat/core/CMakeFiles/compatibility_test.dir/serialization.cpp.o' failed
CMakeFiles/Makefile2:393: recipe for target 'cpp/test/compat/core/CMakeFiles/compatibility_test.dir/all' failed
make[2]: *** [cpp/test/compat/core/CMakeFiles/compatibility_test.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 17%] Building CXX object cpp/test/core/CMakeFiles/core_test_common.dir/capped_allocator_tests_generated/allocator_test_types.cpp.o
[ 17%] Linking CXX static library libcore_test_common.a
[ 19%] Built target core_test_common
make[1]: *** [CMakeFiles/check.dir/rule] Error 2
CMakeFiles/Makefile2:147: recipe for target 'CMakeFiles/check.dir/rule' failed
Makefile:175: recipe for target 'check' failed
make: *** [check] Error 2`